src/query/storage/promremote/storage_unsupported.go

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) 2021  Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package promremote
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/m3db/m3/src/query/block"
+	"github.com/m3db/m3/src/query/storage"
+	"github.com/m3db/m3/src/query/storage/m3/consolidators"
+	"github.com/m3db/m3/src/query/storage/m3/storagemetadata"
+)
+
+func (p *promStorage) FetchProm(
```

Comment:
nit: not quite sure about splitting `promStorage` method definitions across two files, people reading code might find it unexpected

Reply:
I wanted to hide these unimplemented methods to make it easier reading the implemented part.

Reply:
I agree with @vpranckaitis, this is unusual (having some methods for the struct in another file). 
Perhaps it would be more clear to do it like this:
```go
type unimplementedPromStorageMethods struct {}

func (p *unimplementedPromStorageMethods) FetchProm(...
```
And then embed this `unimplementedPromStorageMethods` into the actual struct.

Reply:
Ok, ill just put these unimplemented methods at the end of the source file after a comment.

Reply:
introduced struct, looks pretty neat actually.